### ISSUE TITLE
squoosh crash on static import png

### ISSUE THREAD

**Issue by akomm** [vercel/next.js#26527]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14.15.0

### What browser are you using?

firefox

### What operating system are you using?

linux

### How are you deploying your application?

other

### Describe the Bug

static import of png:
```
Error: `unwrap_throw` failed
    at <anonymous>:wasm-function[168]:0x165c5
    at decode (<anonymous>:wasm-function[7]:0x9e7c)
```
$ grep unwrap_throw -r node_modules
binary file: node_modules/next/dist/next-server/server/lib/squoosh/png/squoosh_png_bg.wasm
binary file: node_modules/next/dist/next-server/server/lib/squoosh/png/squoosh_oxipng_bg.wasm

### Expected Behavior

no error

### To Reproduce

statically import a png file and use the import in jsx/tsx code

**Comment by akomm** [comment]
Might be this: https://github.com/GoogleChromeLabs/squoosh/issues/1057

**Comment by velvetkevorkian** [comment]
I'm also seeing this in Ubuntu in WSL as we upgrade to Next 11.0.1 -- even images that don't throw an error in the build process are showing up with encoding errors in the browser.

As a workaround it looks like you can [disable static images](https://nextjs.org/docs/basic-features/image-optimization#disable-static-imports) in `next.config.js`
```
  images: {
    // https://github.com/vercel/next.js/issues/26527
    disableStaticImages: true,
  },
```
and use [Webpack asset modules](https://webpack.js.org/guides/asset-modules/#resource-assets) instead, e.g. 
```
webpack: (config) => {
    const assetRegex = new RegExp(`.(png|jpe?g|gif|woff|woff2|ico|svg)$`)

    config.module.rules.push({
      test: assetRegex,
      type: 'asset/resource',
      generator: {
        filename: './static/assets/[name]-[contenthash].[ext]',
      },
    })

    return config
  }
```

**Comment by ipy** [comment]
Thanks to @velvetkevorkian 's solution, while using TypeScript I have to use additional loaders to conform the new typings defined for images.

```js
    const assetRegex = new RegExp(`.(png|jpe?g|gif|woff|woff2|ico|svg)$`);
    config.module.rules.push({
      test: assetRegex,
      use: [
        {
          loader: 'string-replace-loader',
          options: {
            search: 'export default (.*)',
            replace: (_, src) =>
              `export default { src: ${src.replace(/;$/, '')} }`,
            flags: 'm',
          },
        },
        {
          loader: 'url-loader',
          options: { outputPath: './static/assets/' },
        },
      ],
      type: 'javascript/auto',
    });
```

```tsx
import img from 'path/to/image.png';

<img src={img.src} />
```

**Comment by dperetti** [comment]
I think it may have to do with the image ratio.

Both jpg and png images below make the build crash.

When using the **jpg image**, the error is:
```
Error: Call retries were exceeded
```
When using the **png image**, the error is:
```
Error: `unwrap_throw` failed
    at <anonymous>:wasm-function[168]:0x165c5
    at encode (<anonymous>:wasm-function[9]:0xbdf5)
```

![nextJsCrash](https://user-images.githubusercontent.com/845612/127991443-da758473-9731-4e5b-a6e3-2df13a4788d5.jpg)
![nextJsCrash](https://user-images.githubusercontent.com/845612/127991448-0e9b5f84-d5a4-46ec-9244-688f415b28eb.png)

**Comment by dperetti** [comment]
[Kudos](https://github.com/vercel/next.js/discussions/27687#discussioncomment-1124135) to @leerob 👍  
```
yarn add next@canary sharp
```
Fixes the issue for me.

**Comment by akomm** [comment]
@dperetti AFAIK it does add the option to use a different optimizer, right? Because I just checked nextjs canary pre-release change-log and did not find explicitly a fix for this. 

I think if it is related to the bug I'v linked [previously](https://github.com/GoogleChromeLabs/squoosh/issues/1057), this can also be fixed by nextjs team without different optimizer by bumping the version of squoosh used. They have fixed the bug it seems.

**Comment by akomm** [comment]
@dperetti AFAIK it adds an option to use an alternative transformer. Because I'v looked through canary pre-release change-logs and so far did not see fix for this in general. Obviously this is nice for those who are blocked by this!

Regarding this issue:

If it is related to the [previously](https://github.com/GoogleChromeLabs/squoosh/issues/1057) linked issue, nextjs team could fix this generally, by bumping squoosh version used.

**Comment by KevinShiCA** [comment]
FYI this seems to be broken on `next@11.1.3-canary.97` again, even with `sharp` installed. Going back to `next@11.1.3-canary.68` as suggested by @jakelazaroff fixes it.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
